vcd/sample_vcd_test_config.json

```diff
@@ -90,6 +90,44 @@
     "uploadPieceSize": 5,
     "uploadProgress": true
   },
+  "testEnvBuild": {
+    "//": "This section is optional for running tests. It's only needed to build the test environment in a new vCD",
+
+    "// ": "Entities that will be created in an empty vCD use names defined above",
+
+    "// ": "vcd.org                    -> organization name",
+    "// ": "vcd.vdc                    -> VDC name",
+    "// ": "vcd.catalog.name           -> catalog name",
+    "// ": "vcd.catalog.catalogItem    -> catalog item name",
+    "// ": "networking.edgeGateway     -> edge gateway name",
+    "// ": "networking.externalNetwork -> external network name",
+
+    "//": "Values used to build a new environment. Build will fail if the next 5 ones are empty",
+    "gateway": "10.15.21.1",
+    "netmask": "255.255.224.0",
+    "externalNetworkStartIp": "10.15.21.10",
+    "externalNetworkEndIp": "10.15.21.11",
+    "dns1": "10.15.21.101",
+    "dns2": "8.8.4.4",
+
+    "//": "external network may be different from the one defined for testing",
```

Comment:
Please also add a comment above in this JSON above the these three fields on what are they used for:

```
    "vcenter" : "vC1",
    "externalNetworkPortGroup": "ForTestingPG",
    "externalNetworkPortGroupType": "DV_PORTGROUP"
```

Reply:
p.s. Now I see that this really needs some clarification, because I used same values for port group and type in the two places, which led to `make testacc` failures down the road:

```
--- FAIL: TestAccVcdEdgeGatewayComplex (7.13s)
    testing.go:568: Step 0 error: errors during apply:

        Error: error creating external network: &errors.errorString{s:"error instantiating a new ExternalNetwork: API Error: 400: [ a8756f4c-8bdb-4e82-ac80-f9c1f2d91ec6 ] Cannot associate portgroup to External Network. Portgroup is in use: true"}

          on /var/folders/0g/mmbktw7j0vl9d1hhrxbpg05r0000gq/T/tf-test231292695/main.tf line 3:
          (source code not available)
```

Reply:
> Please also add a comment above in this JSON above the these three fields on what are they used for:
```
>     "vcenter" : "vC1",
>     "externalNetworkPortGroup": "ForTestingPG",
>     "externalNetworkPortGroupType": "DV_PORTGROUP"
```

These values are used for external network testing. There is already a note saying that you can use different values if you want your external network (which will be created and will persist) be different than the one for testing (which will be created and destroyed).

Reply:
> p.s. Now I see that this really needs some clarification, because I used same values for port group and type in the two places, which led to `make testacc` failures down the road:
> 
> ```
> --- FAIL: TestAccVcdEdgeGatewayComplex (7.13s)
>     testing.go:568: Step 0 error: errors during apply:
> 
>         Error: error creating external network: &errors.errorString{s:"error instantiating a new ExternalNetwork: API Error: 400: [ a8756f4c-8bdb-4e82-ac80-f9c1f2d91ec6 ] Cannot associate portgroup to External Network. Portgroup is in use: true"}
> 
>           on /var/folders/0g/mmbktw7j0vl9d1hhrxbpg05r0000gq/T/tf-test231292695/main.tf line 3:
>           (source code not available)
> ```

As I mention above, the expectation is that users know how to configure this file for testing, and then add information to build up the environment.
Using the same information for both testing and build-up will obviously lead to a clash: the external network in the build-up is the one that will be permanent, while the external network in the tests will be created using a different port group and later destroyed. Using the same port group for both will lead to an error.